### PR TITLE
W-15921577 Java Module Files to edit

### PR DESCRIPTION
**Pull request by valkyrie69**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: connectors-home/modules/ROOT/partials/acb-java-module-partial-edit.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//Used in the configure-java-module-acb file for Java Module configuration in ACB
+
+// tag::advanced-tab-attributes[]
+. Select the *Advanced* tab to configure the following attributes: 
++
+[%header%autowidth.spread]
+|===
+| UI Field | Description | Required
+| Output MIME type | MIME type of the payload this component outputs.| No
+| Parameters | Appends a key-value pair for a reader property to the value of the Output MIME attribute. Multiple key-value pairs are allowed.| No
```
Comment (hannanelson) [inline connectors-home/modules/ROOT/partials/acb-java-module-partial-edit.adoc:10]: Any way to make the descriptions parallel?

Comment (hannanelson) [inline connectors-home/modules/ROOT/partials/acb-java-module-partial-edit.adoc:10]: Output MIME type attribute?

File: connectors-home/modules/ROOT/partials/acb-java-module-partial-edit.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+| Output encoding | Encoding of the payload this component outputs. | No
```
Comment (hannanelson) [inline connectors-home/modules/ROOT/partials/acb-java-module-partial-edit.adoc:11]: I wonder if there's a way to make this description more clear? I had to read it twice.
Payload encoding for this component's output? 
or maybe it just needs a "that"?

File: connectors-home/modules/ROOT/partials/acb-java-module-partial-edit.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+| Target variable | Name of the variable that stores the operation's output. | No
+| Target value | Expression that evaluates the operation's output. The expression outcome is stored in the target variable. | No
+| Error Mappings a| Creates custom error types that you map to Mule error types. The mappings can help you further differentiate and group errors in your app. This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:mule-runtime::mule-error-concept.adoc[]. | No
```
Comment (hannanelson) [inline connectors-home/modules/ROOT/partials/acb-java-module-partial-edit.adoc:14]: Any way to combine the third and fourth sentences (and put a positive spin) (and use contraction)?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 4]
```diff
@@ -0,0 +1,331 @@
+= Configure the Java Module Using Anypoint Code Builder
+
+Use the Java module in your Anypoint Code Builder project to define custom Java code. To use Anypoint Code Builder to configure the Java module, create a basic integration project in Anypoint Code Builder, add connectors and other components to your Mule application to process your business logic, and then configure the attributes for the Java Module. 
+
+== Before You Begin
+
+Before creating an Anypoint Code Builder integration project, you must:
+
+* xref:anypoint-code-builder::start-acb.adoc[Set up and access the Anypoint Code Builder web or desktop IDE]
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
* xref:anypoint-code-builder::start-acb.adoc[Set up and access the Anypoint Code Builder web or desktop IDE].
```

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+* Have credentials to access Anypoint Platform
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
* Have credentials to access Anypoint Platform.
```

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+* Ensure that the Extension Pack for Java (from Microsoft) is installed in Visual Studio Code.
+
+[[creating-new-integration-project]]
+== Creating a New Integration Project
+
+include::connectors::partial$configure-acb-connectors-partial.adoc[tag="acb-create-integration-project"]
+
+When you create a new Anypoint Code Builder project, `src/main/java` and `/src/test/java` folders are created in the project. For existing projects, you can manually create these folders.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
When you create a new integration project, Anypoint Code Builder creates the `src/main/java` and `/src/test/java` folders in the project. For existing projects, you can manually create these folders. 
```

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+=== Configuring the New Operation
+
+Create a new class instance by using the *New* operation of the Java module. The identifier of a constructor includes the class and constructor names, and the constructor has a full description of its signature, including the types of each parameter.
+
+For example, if you invoke the constructor `Me(String name, int age)`, which belongs to the `org.bar.Me` class, then the identifier of the method is `Me(String,int)`.
+
+To configure the attributes for *New* in the Anypoint Code Builder UI:
+
+. Click the *New* component on the canvas.
+. In the *General* tab:
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
. On the *General* tab:
```

Comment (hannanelson) [reply java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: I don't see this specifically called out in the SR, but it is used in examples.

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+.. In *Class*, search and select the Java class that contains the method to use.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: Is Class a section?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+For example, if you invoke the constructor `Me(String name, int age)`, which belongs to the `org.bar.Me` class, then the identifier of the method is `Me(String,int)`.
+
+To configure the attributes for *New* in the Anypoint Code Builder UI:
+
+. Click the *New* component on the canvas.
+. In the *General* tab:
+.. In *Class*, search and select the Java class that contains the method to use. 
++
+You can search for all the available Java classes for your project, including classes you create and native Java classes.  
+.. In *Constructor*, search and select the constructor to use.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: Same q as line 49.

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+To configure the attributes for *New* in the Anypoint Code Builder UI:
+
+. Click the *New* component on the canvas.
+. In the *General* tab:
+.. In *Class*, search and select the Java class that contains the method to use. 
++
+You can search for all the available Java classes for your project, including classes you create and native Java classes.  
+.. In *Constructor*, search and select the constructor to use. 
++
+NOTE: If this is blank, click the refresh icon to populate it.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: This field? 

https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-%22it,%22%22this,%22%22these%22

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+. Select the *Advanced* tab to configure the following attributes: 
++
+[%header%autowidth.spread]
+|===
+| UI Field | Description | Required
+| Target variable | Name of the variable to assign the class instance to. | No
+| Target value | Expression that evaluates the operation's output. The expression outcome is stored in the target variable. | No
+| Error Mappings a| Creates custom error types that you map to Mule error types. The mappings can help you further differentiate and group errors in your app. This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:mule-runtime::mule-error-concept.adoc[]. | No
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: Parallelism? See also line 14. If these are identical, can you use a partial?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 12]
```diff
@@ -0,0 +1,331 @@
...
+
+The *Invoke* operation invokes methods with the provided arguments using a specified class instance. The identifier of the method to invoke includes the class and method names.
+
+To configure the attributes for *Invoke* in the Anypoint Code Builder UI:
+
+. Click the *Invoke* component on the canvas.
+. In the *General* tab:
+.. Select the class *Instance* for invoking the method.
++
+This is the target variable assigned to the *New* operation.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: This class?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+
+[[configuring-invoke]]
+=== Configuring the Invoke Operation
+
+The *Invoke* operation invokes methods with the provided arguments using a specified class instance. The identifier of the method to invoke includes the class and method names.
+
+To configure the attributes for *Invoke* in the Anypoint Code Builder UI:
+
+. Click the *Invoke* component on the canvas.
+. In the *General* tab:
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: On

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc (hunk 11 above)
Comment (hannanelson) [reply java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: Oh, maybe you defined the partials but didn't use them?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 14]
```diff
@@ -0,0 +1,331 @@
...
+.. Select the class *Instance* for invoking the method.
++
+This is the target variable assigned to the *New* operation.
+.. In *Class*, select the Java class that contains the method to use.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: Different from line 49?
If not, create and use partials for these duplicated steps?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 15]
```diff
@@ -0,0 +1,331 @@
...
+. In the *General* tab:
+.. Select the class *Instance* for invoking the method.
++
+This is the target variable assigned to the *New* operation.
+.. In *Class*, select the Java class that contains the method to use. 
++
+You can search for all the available Java classes for your project, including classes you create and native Java classes.  
+.. In *Method*, search and select the method to use. 
++
+NOTE: If this is blank, click the refresh icon to populate it.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: this field?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 16]
```diff
@@ -0,0 +1,331 @@
...
+
+[[configuring-invoke-static]]
+=== Configuring the Invoke Static Operation
+
+Invoke static methods with the arguments you provide. 
+
+To configure the attributes for *Invoke Static* in the Anypoint Code Builder UI:
+
+. Click the *Invoke Static* component on the canvas.
+. In the *General* tab:
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: On

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 17]
```diff
@@ -0,0 +1,331 @@
...
+.. In *Class*, select the Java class that contains the method to use.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: make this a partial?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 18]
```diff
@@ -0,0 +1,331 @@
...
+. Click the *Invoke Static* component on the canvas.
+. In the *General* tab:
+.. In *Class*, select the Java class that contains the method to use. 
++
+You can search for all the available Java classes for your project, including classes you create and native Java classes.  
+.. In *Method*, search and select the method to use. 
++
+Only the static methods from the class are displayed.
++
+NOTE: If this is blank, click the refresh icon to populate it.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: this field?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 19]
```diff
@@ -0,0 +1,331 @@
...
+
+. Click the *Invoke* component on the canvas.
+. In the *General* tab:
+.. Select the class *Instance* for invoking the method.
++
+This is the target variable assigned to the *New* operation.
+.. In *Class*, select the Java class that contains the method to use. 
++
+You can search for all the available Java classes for your project, including classes you create and native Java classes.  
+.. In *Method*, search and select the method to use.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: (See previous comments on the same text.)

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 20]
```diff
@@ -0,0 +1,331 @@
...
+.. In *Method*, search and select the method to use. 
++
+Only the static methods from the class are displayed.
++
+NOTE: If this is blank, click the refresh icon to populate it.
+include::connectors::partial$acb-java-module-partial.adoc[tag="advanced-tab-attributes"]
+
+=== Configuring the Validate Type Operation
+
+Use this operation to validate that the instance is an `instanceof` the expected class.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
Use the Valid Type operation to validate that the instance is an `instanceof` the expected class:
```

Comment (hannanelson) [reply java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: In general, we don't use elements this way.
https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-UIelements

Suggest 

If you need to keep it, suggest something like:
Use this operation to validate that the instance is an instance of (`instanceof`) the expected class.

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 21]
```diff
@@ -0,0 +1,331 @@
...
+
+. Click the *Validate type* component on the canvas.
+. In the *General* tab:
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: On. I'll stop marking these, yah?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 22]
```diff
@@ -0,0 +1,331 @@
...
+.. Select the *Class* the instance belongs to.
+.. Enter the name of the class *Instance* to validate.
+.. Select *Accept subtypes* to accept subtypes of a class. 
++
+Unselect this option if the class instance has to be of the exact same class.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
Deselect this option if the class instance must be of the same class. 
```

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 23]
```diff
@@ -0,0 +1,331 @@
...
+    public Boolean isTaxFree(Double price) {
+        if (price < 10) {
+            return true;
+        }
+        return false;
+    }
+}
+---- 
++
+In this example code, there are two methods:
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: You could use code callouts for this instead (and avoid the "there are" construction):
`[calloutlist]`

See https://confluence.internal.salesforce.com/display/MTDT/Formatting+with+Asciidoc#FormattingwithAsciidoc-AddCalloutstoCodeBlocks

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 24]
```diff
@@ -0,0 +1,331 @@
...
+* `calculateTax`, which receives the `price` and `percentAdditionalTax` parameters.
+* `isTaxFree`, which receives the `price` parameter.
+
+=== Configure an HTTP Trigger
+
+. Click the *Listener* component on the canvas UI.
+. (Optional) Click the edit icon to change the name of the *Listener* component.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: Kim's guidance says to avoid.
https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-optionalsteps

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 25]
```diff
@@ -0,0 +1,331 @@
...
++
+In this example code, there are two methods:
+* `calculateTax`, which receives the `price` and `percentAdditionalTax` parameters.
+* `isTaxFree`, which receives the `price` parameter.
+
+=== Configure an HTTP Trigger
+
+. Click the *Listener* component on the canvas UI.
+. (Optional) Click the edit icon to change the name of the *Listener* component. 
+. If you created a configuration file for the connector, select it, or add the HTTP configuration to the XML, outside of the flow, for example:
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: Hard to parse. Maybe something can go in parentheses?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 26]
```diff
@@ -0,0 +1,331 @@
...
+	http://www.mulesoft.org/schema/mule/java http://www.mulesoft.org/schema/mule/java/current/mule-java.xsd">
+  <http:listener-config name="HTTP_listener_config">
+    <http:listener-connection host="0.0.0.0" port="8081" />
+  </http:listener-config>
+  <flow name="name1">
+    <http:listener path="/calculatePath" config-ref="HTTP_listener_config" doc:name="Listener" doc:id="511e4743-a5ed-4e74-8174-ead32ff575ea" allowedMethods="POST" />
+  </flow>
+</mule>
+----
+. In the *General* tab, in the *Path* field: `/calculateTax`.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
. On the *General* tab, enter `/calculateTax` in the *Path* field.
```

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 27]
```diff
@@ -0,0 +1,331 @@
...
+. In the *Advanced* tab, set the *Allowed methods* to `POST`.
+
+=== Create a New Class Instance
+
+Since the method used in this example is non-static, you must create a new class instance:
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
Because the method used in this example is non-static, you must create a new class instance:
```

Comment (hannanelson) [reply java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-synonymsandwordswithmultiplemeanings

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 28]
```diff
@@ -0,0 +1,331 @@
...
+
+. Click the *New* component on the canvas.
+. In the *General* tab:
+.. Select the *Class* that contains the method to use, in this case, `com.me.TaxCalculator`.
+.. Select the *Method* to use: `TaxCalculator()`.
++
+NOTE: If this is blank, click the refresh icon to populate it.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: this field?

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 29]
```diff
@@ -0,0 +1,331 @@
...
+----
+
+=== Add an Invoke Operation
+
+To call your method, add an *Invoke* operation:
+
+. Click the *Invoke* component on the canvas.
+. In the *General* tab:
+.. Enter `#[vars.taxCalculator]` as the class *Instance* for invoking the method.
+.. Select the *Class* that contains the method to use, in this case ,`com.me.TaxCalculator`.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
.. Select the *Class* that contains the method to use, in this case, `com.me.TaxCalculator`.
```

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 30]
```diff
@@ -0,0 +1,331 @@
...
+.. Select the *Method* to use: `calculateTax(java.lang.Double,java.lang.Integer)`.
++
+The list of methods shown are standard non-static Java methods in addition to the methods you created.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
The methods shown are the standard non-static Java methods and the methods you created. 
```

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 31]
```diff
@@ -0,0 +1,331 @@
...
+        arg1: payload.percentAdditionalTax
+      }]</java:args>
+    </java:invoke>
+  </flow>
+</mule>
+----
+
+=== Add the Set Payload Transformer
+
+The Set Payload (set-payload) component lets you update the payload of the message. The payload can be a literal string or a DataWeave expression.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: font for "set-payload"?

Comment (hannanelson) [reply java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: User value first? To update the payload...

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 32]
```diff
@@ -0,0 +1,331 @@
...
+
+. In the canvas UI, click the image:anypoint-code-builder::icon-plus.png["",15,15] (*Add component*) icon.
+. In the *Add Component* panel, search for `Set Payload` and select it. 
+. Click the *Set Payload* component. 
+. In the *General* tab enter `#[vars.taxResult as String]` for the *Value*.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]:
```suggestion
. In the *General* tab, enter `#[vars.taxResult as String]` for the *Value*.
```

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc [hunk 33]
```diff
@@ -0,0 +1,331 @@
+= Configure the Java Module Using Anypoint Code Builder
+
+Use the Java module in your Anypoint Code Builder project to define custom Java code. To use Anypoint Code Builder to configure the Java module, create a basic integration project in Anypoint Code Builder, add connectors and other components to your Mule application to process your business logic, and then configure the attributes for the Java Module.
```
Comment (hannanelson) [inline java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: Put user value first? To define custom code...

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc (hunk 8 above)
Comment (valkyrie69) [reply java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: No, it's an input box

File: java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc (hunk 9 above)
Comment (valkyrie69) [reply java/1.2/modules/ROOT/pages/configure-java-module-acb-edit.adoc]: Input box

**Review by sathya0 — APPROVED** [review]
